### DISCUSSION TITLE
atomWithStorage in multiple Providers

### DISCUSSION THREAD

**Discussion by miquik** [pmndrs/jotai#2196] [General]
Hello everyone,
I'm trying to solve a problem with atom, but I can't figure it out. 
I have an atom that encloses a `settings ` object. This atom is used in a reusable component (in which a `Provider ` is defined so that each "instance" of the component has its own `settings` values).

No problem so far: now I want some properties** of the `settings` to be stored in `localStorage`; so this storage should be done per Provider. How can I achieve this behaviour?

I hope I have managed to explain myself. 

** Note: I mean that if for example `settings = { name: 'foo', lastname: 'last' }`, I can decide to store only the `lastname ` property in `localStorage`.

Thanks

**Comment by DavidJFelix** [comment]
Well, `localstorage` and by extension `atomWithStorage` using localstorage will **always** be global, regardless of `jotai`'s `Provider`'s `store` or `jotai-scope`'s  `Scope` or `bunshi`'s `Scope`. If you pull a value using the same key it will always be globally the same value.

So you have a few options.

* Use a unique `localstorage` key per settings object
* Use the same key for `localstorage` and nest settings objects inside that key using unique keys for each settings object.

Either of these are going to involve writing your own atom and possibly storing atoms within atoms based on `Provider` or `Scope`.

**Reply by miquik** [reply]
Thank you for the answer, but I have some difficulty visualising how 'atom in atom' can help me here, I still don't understand what it is for

**Reply by DavidJFelix** [reply]
I haven't advocated for atom in atom here -- I don't think that will help you it'll just be the same problem. I think you need to make a decision on how to co-locate localized storage in a global store by choosing one of the above bullet points and designing from there. Once you make a choice for how to store these settings I think how you do it is much more straight forward.
